src/java/org/apache/cassandra/net/async/RebufferingByteBufDataInputPlus.java

```diff
@@ -36,6 +37,11 @@
 
 public class RebufferingByteBufDataInputPlus extends RebufferingInputStream implements ReadableByteChannel
 {
+    /**
+     * Default to a very large value.
+     */
+    private static final long DEFAULT_REBUFFER_BLOCK_IN_MILLIS = TimeUnit.DAYS.toMillis(2);
```

Comment:
The corresponding pre-4.0 code just does blocking IO on a socket, with no timeouts. Thus it blocks forever. 2 days seems shorter than forever, but I'm game to change this to any time-bounded value.

Reply:
I think it doesn't make sense to wait for more than 5 minutes to rebuffer. Given buffer sizes are on the order of a few megabytes at most, 5 minutes is an eternity. It's best to keep it short so failures are exposed sooner than later and rather than threads being stuck waiting on a call that doesn't timeout in a reasonable amount of time.

Reply:
I've lowered it to 3 minutes.